pkg/drivers/kic/node/node.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/machine/libmachine/state"
+	"k8s.io/minikube/pkg/drivers/kic/oci"
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/command"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// Docker default bridge network is named "bridge" (https://docs.docker.com/network/bridge/#use-the-default-bridge-network)
+	DefaultNetwork  = "bridge"
+	ClusterLabelKey = "io.x-k8s.kind.cluster" // ClusterLabelKey is applied to each node docker container for identification
+	NodeRoleKey     = "io.k8s.sigs.kic.role"
+	DefaultOci      = "docker"
+)
+
+// Node represents a handle to a kic node
+// This struct must be created by one of: CreateControlPlane
+type Node struct {
+	// must be one of docker container ID or name
+	name string
+	// cached node info etc.
+	cache *nodeCache
+	R     command.Runner // Runner
+}
+
+// WriteFile writes content to dest on the node
+func (n *Node) WriteFile(dest, content string, perm string) error {
+	// create destination directory
+	cmd := exec.Command("mkdir", "-p", filepath.Dir(dest))
+	rr, err := n.R.RunCmd(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create directory %s cmd: %v output:%q", cmd.Args, dest, rr.Output())
+	}
+
+	cmd = exec.Command("cp", "/dev/stdin", dest)
+	cmd.Stdin = strings.NewReader(content)
+
+	if rr, err := n.R.RunCmd(cmd); err != nil {
+		return errors.Wrapf(err, "failed to run: cp /dev/stdin %s cmd: %v output:%q", dest, cmd.Args, rr.Output())
+	}
+
+	cmd = exec.Command("chmod", perm, dest)
+	_, err = n.R.RunCmd(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to run: chmod %s %s", perm, dest)
+	}
+	return nil
+}
+
+// IP returns the IP address of the node
+func (n *Node) IP() (ipv4 string, ipv6 string, err error) {
+	// use the cached version first
+	cachedIPv4, cachedIPv6 := n.cache.IP()
+	if cachedIPv4 != "" && cachedIPv6 != "" {
+		return cachedIPv4, cachedIPv6, nil
+	}
+	// retrieve the IP address of the node using docker inspect
+	lines, err := oci.Inspect(n.name, "{{range .NetworkSettings.Networks}}{{.IPAddress}},{{.GlobalIPv6Address}}{{end}}")
```

Comment:
Can we remove caching from the first iteration of this PR? This code doesn't seem to address cache invalidation, and I'm very wary to commit to accepting the complexity of a cache without first weighing it against the performance benefits.